queries/pandas/q5.py

```diff
@@ -42,21 +42,24 @@ def query() -> pd.DataFrame:
         supplier_ds = supplier_ds()
 
         rsel = region_ds.r_name == "ASIA"
-        osel = (orders_ds.o_orderdate >= date1) & (orders_ds.o_orderdate < date2)
```

Comment:
Filters before the join is not a direct translation from SQL. The optimizer has to determine which predicates can pass which joins.